src/main/java/org/apache/datasketches/kll/KllDoublesSketch.java

```diff
@@ -336,12 +363,98 @@ public void update(final double item, final long weight) {
     if (Double.isNaN(item)) { return; } //ignore
     if (readOnly) { throw new SketchesArgumentException(TGT_IS_READ_ONLY_MSG); }
     if (weight < 1L) { throw new SketchesArgumentException("Weight is less than one."); }
-    if (weight == 1L) { KllDoublesHelper.updateDouble(this, item); }
-    else { KllDoublesHelper.updateDouble(this, item, weight); }
+    if (weight == 1L) { updateDouble(this, item); }
+    else {
+      if (weight < levelsArr[0]) {
+        for (int i = 0; i < (int)weight; i++) { updateDouble(this, item); }
+      } else {
+        final KllHeapDoublesSketch tmpSk = new KllHeapDoublesSketch(getK(), DEFAULT_M, item, weight);
+        merge(tmpSk);
+      }
+    }
     doublesSV = null;
   }
 
-  //restricted
+  // VECTOR UPDATE
+
+  /**
+   * Vector update. Updates this sketch with the given array (vector) of items, starting at the items
+   * offset for a length number of items. This is not supported for direct sketches.
+   * @param items the vector of items
+   * @param offset the starting index of the items[] array
+   * @param length the number of items
+   */
+  public void update(final double[] items, final int offset, final int length) {
+    if (readOnly) { throw new SketchesArgumentException(TGT_IS_READ_ONLY_MSG); }
+    if (length == 0) { return; }
+    if (!hasNaN(items, offset, length)) {
+      updateDouble(items, offset, length); //fast path
+      return;
```

Comment:
should this path invalidate the sorted view?

Reply:
Ooh! Nice catch!  will fix.